### PR TITLE
Added Giphy integration

### PR DESCRIPTION
**Pull request by dorsha**
Giphy API details: https://github.com/Giphy/GiphyAPI  

Usage examples:
```!giphy hello``` - Returns random GIF for 'hello'
```!giphy hello as-json=true``` - Returns JSON response for 'hello'

image

klkchhy6ef

### CODE REVIEW

**Comment by rishibhargava** [comment]
What category is the giffy integration in :)? Love it

________________________________

**Comment by dorsha** [comment]
We need to add new category I guess
I put it in communication :)

File: Integrations/integration-Giphy.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  name: apikey
+  defaultvalue: ""
+  type: 4
+  required: true
+script:
+  script: |-
+    var sendRequest = function(tag) {
+        var serverUrl = params.url;
+        var apiKey = params.apikey;
+        var requestUrl = params.url + '?api_key=' + apiKey + '&tag=' + tag;
```
Comment (aviadl) [inline Integrations/integration-Giphy.yml]: you need to sanitize the tag for special characters 
We have a function for this

File: Integrations/integration-Giphy.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  required: true
+- display: API Key
+  name: apikey
+  defaultvalue: ""
+  type: 4
+  required: true
+script:
+  script: |-
+    var sendRequest = function(tag) {
+        var serverUrl = params.url;
```
Comment (aviadl) [inline Integrations/integration-Giphy.yml]: Need to handle case where url param ends with a '/'

File: Integrations/integration-Giphy.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+detaileddescription: 'You can get API Key from: https://github.com/Giphy/GiphyAPI#access-and-api-keys'
+configuration:
+- display: Giphy URL
+  name: url
+  defaultvalue: http://api.giphy.com/v1/gifs/random
+  type: 0
+  required: true
+- display: API Key
+  name: apikey
+  defaultvalue: ""
```
Comment (aviadl) [inline Integrations/integration-Giphy.yml]: Do you want to add the default test api key ?
Not sure about it, but than users can use it easily :)

Comment (aviadl) [inline Integrations/integration-Giphy.yml]: I think so, it is public ...
lets ask Dan

File: Integrations/integration-Giphy.yml (hunk 2 above)
Comment (dorsha) [inline Integrations/integration-Giphy.yml]: Fixed

**Comment by dorsha** [comment]
@aviadl - I fixed everything, please take a look and approve.

**Pull request merged**